src/C-interface/dense/bml_normalize_dense_typed.c

```diff
@@ -69,40 +69,68 @@ void *TYPED_FUNC(
 
     double *eval = bml_allocate_memory(sizeof(double) * 2);
 
-#pragma omp parallel for \
+#ifdef _OPENMP
+    int num_threads = omp_get_num_threads();
+#else
+    int num_threads = 1;
+#endif
+
+    double _emin[num_threads];
+    double _emax[num_threads];
+
```

Comment:
We avoid the `critical` section by creating an array of min/max values we can share safely across the threads.